jina/hubble/hubio.py

```diff
@@ -61,62 +86,161 @@ def push(self) -> None:
                 self.logger.info(
-                    'You can use this Executor in the Flow via '
-                    + colored(usage, 'cyan', attrs='underline')
+                    "You can use this Executor in the Flow via "
+                    + colored(usage, "cyan", attrs="underline")
                 )
 
-            else:
+            elif resp.text:
+                # NOTE: sometimes resp.text returns empty
                 raise Exception(resp.text)
+            else:
+                resp.raise_for_status()
 
         except Exception as e:  # IO related errors
             self.logger.error(
-                f'Error when trying to push the executor at {self.args.path}: {e!r}'
+                f"Error when trying to push the executor at {self.args.path}: {e!r}"
+            )
+
+    def fetch(self, id: str, tag: str = None) -> HubExecutor:
+        """Fetch the executor meta info from Jina Hub.
+        :param id: the ID of the executor
+        :param tag: the version tag of the executor
+        :return: meta of executor
+        """
+        import requests
+
+        pull_url = JINA_HUBBLE_PUSHPULL_URL + f"/{id}"
+        path_params = []
+        if self.args.secret:
+            path_params.append(f"secret={self.args.secret}")
+        if tag:
+            path_params.append(f"tag={tag}")
+
+        if len(path_params) > 0:
+            pull_url += "?" + ",".join(path_params)
+
+        resp = requests.get(pull_url)
+        if resp.status_code != 200:
+            if resp.text:
+                raise Exception(resp.text)
+            else:
+                resp.raise_for_status()
+
+        resp = resp.json()
+        assert resp["id"] == id
+
+        result = HubExecutor(
+            resp["id"],
+            f'v{resp["currentVersion"]}',
+            resp["visibility"],
+            resp["pullPath"],
+            resp["package"]["download"],
+            resp["package"]["md5"],
+        )
+
+        return result
+
+    def pull(self) -> None:
+        """Pull the executor pacakge from Jina Hub."""
+        cached_zip_file = None
+        try:
+            id = self.args.id
+            tag = None
+
+            executor = self.fetch(id, tag)
+
+            tag = executor.current_tag
+
+            image_name = executor.image_name
+            archive_url = executor.archive_url
+            md5sum = executor.md5sum
+
+            if self.args.docker:
+                # pull the Docker image
+                with TimeContext(f"pulling {image_name}", self.logger):
+                    image = self._client.images.pull(image_name)
+                if isinstance(image, list):
+                    image = image[0]
+                image_tag = image.tags[0] if image.tags else ""
+                self.logger.success(
+                    f'🎉 pulled {image_tag} ({image.short_id}) uncompressed size: {get_readable_size(image.attrs["Size"])}'
+                )
+            else:
+                if exist_locall(id, tag):
```

Comment:
The overlong `if...else...` is error-prone. 
```suggestion
                self.logger.success(
                    f'🎉 pulled {image_tag} ({image.short_id}) uncompressed size: {get_readable_size(image.attrs["Size"])}'
                )
                return
            if exist_locall(id, tag):
```